### ISSUE TITLE
bug: Draft got lost, when having deployed and draft version in edit view in parallel

### ISSUE THREAD

**Issue by wasnertobias** [windmill-labs/windmill#3646]
### Describe the bug

Had the deployed version in edit mode (as the non-edit mode shows too less detail) and a draft version concurrently in another browser Tab.

After I made changes to a subflow of the draft (Input Data changed) I had to relead the draft version, otherwise Windmill does not recognize the change Input Data for the subflow. So before I did reload the draft version I clicked on "Save Draft". It told me that there are no current changes, it offered me to "save anyway", which I did not find a reason to click on. So I reloaded.

Badumm, the draft is gone, very nice. It just loads the deployed version without the usual pop-up in the upper right corner to load the deployed version.

![Screenshot 2024-04-30 at 15 43 01](https://github.com/windmill-labs/windmill/assets/1354793/60fa1619-bd28-4e28-9067-f87a4f3a2d16)

### To reproduce

Described above

### Expected behavior

Draft is not deleted for no aparent reason.

### Browser information

Chromium Version 124.0.6367.92

### Application version

EE v1.320.3-4-g39e1ea3d5

**Comment by rubenfiszel** [comment]
yes the issue is because we use localStorage which is across tabs. We need to migrate to use anchor to persist local changes or use other means to work well with multiple tabs on the same flow

**Comment by wasnertobias** [comment]
It is not even separated between workspaces, as I just found out. I edited something on PROD due to the following:
- https://github.com/windmill-labs/windmill/issues/3677#issuecomment-2116961173
- https://github.com/windmill-labs/windmill/issues/3647

Then I noticed and switched to STAGING workspace: Now it asks via toast whether I want to keep my changes, although I never made any in STAGING.
